wremnants/theory_corrections.py

```diff
@@ -703,110 +718,71 @@ def make_pdf_uncertainty_helper_by_helicity(
     # load helicity cross sections from file
     with h5py.File(filename, "r") as h5file:
         results = input_tools.load_results_h5py(h5file)
-        if proc == "Z":
-            if f"nominal_gen_{pdf}" not in results["ZmumuPostVFP"]["output"].keys():
+        proc_map = {
+            "Z": ("ZmumuPostVFP",),
+            "W": ("WplusmunuPostVFP", "WminusmunuPostVFP"),
+        }
+
+        def _collect_pdf_hist(pdf_name):
+            hist_key = f"nominal_gen_{pdf_name}"
+            hists = []
+            for output_key in proc_map.get(proc, ()):
+                outputs = results[output_key]["output"]
+                if hist_key not in outputs:
+                    logger.warning(
+                        f"Did not find {pdf_name} in {filename}. Not creating histogram of PDF variations by helicities for this set."
+                    )
+                    return None
+                hists.append(outputs[hist_key].get())
+            if not hists:
                 logger.warning(
-                    f"Did not find PDF set {pdf} in {filename}. Not creating histogram of PDF variations by helicities for this set."
+                    f"Process {proc} is not supported when building PDF variations."
                 )
                 return None
-            pdf_vars = results["ZmumuPostVFP"]["output"][f"nominal_gen_{pdf}"].get()
-        elif proc == "W":
-            if f"nominal_gen_{pdf}" not in results["WplusmunuPostVFP"]["output"].keys():
-                logger.warning(
-                    f"Did not find PDF set {pdf} in {filename}. Not creating histogram of PDF variations by helicities for this set."
-                )
+            combined = hists[0]
+            for hist_component in hists[1:]:
+                combined = hh.addHists(combined, hist_component)
+            return combined
+
+        pdf_vars = _collect_pdf_hist(pdf)
+        if pdf_vars is None:
+            return None
+
+        if pdf_renorm == pdf:
+            pdf_renorm = pdf_vars
+        else:
+            pdf_renorm_hist = _collect_pdf_hist(pdf_renorm)
+            if pdf_renorm_hist is None:
                 return None
-            pdf_vars_Wp = results["WplusmunuPostVFP"]["output"][
-                f"nominal_gen_{pdf}"
-            ].get()
-            pdf_vars_Wm = results["WminusmunuPostVFP"]["output"][
-                f"nominal_gen_{pdf}"
-            ].get()
-            pdf_vars = hh.addHists(pdf_vars_Wp, pdf_vars_Wm)
+            pdf_renorm = pdf_renorm_hist
 
     # construct the correction tensor
     corr_ax = hist.axis.Boolean(name="corr")
     vars_ax = pdf_vars.axes[var_ax_name]
     axes_no_scale = pdf_vars.axes[:-1]
-    corr_coeffs = hist.Hist(*axes_no_scale, corr_ax, vars_ax)
+    if central_weights:
+        new_vars_ax = hist.axis.StrCategory(["nominal"], name="vars")
+        corr_coeffs = hist.Hist(*axes_no_scale, corr_ax, new_vars_ax)
+    else:
+        corr_coeffs = hist.Hist(*axes_no_scale, corr_ax, vars_ax)
```

Comment:
can save one line by removing the line above with ```vars_ax = pdf_vars.axes[var_ax_name]``` and doing
```
if central_weights:
    vars_ax = hist.axis.StrCategory(["nominal"], name="vars")
else:
    vars_ax = pdf_vars.axes[var_ax_name]
corr_coeffs = hist.Hist(*axes_no_scale, corr_ax, vars_ax)
```

Reply:
vars_ax is needed later to overwrite the axis in the correction tensor